locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,264 @@
+                 ^       |                                                 |
+                 |       ^   emit .pause();        +=================+     |
+                 |       ^---^---------------------+  return false;  <-----+---+
+                 |                                 +=================+         |
+                 |                                                             |
+                 ^   when queue is empty   +============+                      |
+                 ^---^-----------------^---<  Buffering |                      |
+                     |                     |============|                      |
+                     +> emit .drain();     |  <Buffer>  |                      |
+                     +> emit .resume();    +------------+                      |
+                                           |  <Buffer>  |                      |
+                                           +------------+  add chunk to queue  |
+                                           |            <--^-------------------<
+                                           +============+
+```
+
+
+_Note_: The `.pipe()` function is typically where backpressure is invoked. In 
+an isolate application, both [`Readable`][] and [`Writable`][] streams 
+should be present. If you are writing an application meant to accept a 
+[`Readable`][] stream, or pipes to a [`Writable`][] stream from another app, 
+you may omit this detail.
+
+## Example App
+
+Since [Node.js v0.10][], the [`Streams`][] class has offered the ability to 
+overwrite the functionality of [`.read()`][] or [`.write()`][] by using the
+underscore version of these respective functions ([`._read()`][] and 
+[`._write()`][]).
+
+There are guidelines documented for [implementing Readable streams][] and 
+[implementing Writable streams][]. We will assume you've read these over. 
+
+This application will do something very simple: take the data source and 
+perform something fun to the data, and then write it to another file.
+
+## Rules to Abide For Writing Custom Streams 
+
+Recall that a [`.write()`][] may return true or false dependent on some 
+conditions. Thus, when building our own [`Writable`][] stream, we must pay 
+close attention those conditions and the return value:
+
+* If the write queue is busy, [`.write()`][] will return false.
+* If the data chunk is too large, [`.write()`][] will return false (the limit 
+is indicated by the variable, [`highWaterMark`][]).
+
+_Note_: In most machines, there is a byte size that is determines when a buffer 
+is full (which will vary across different machines). Node.js allows you to set 
+your own custom [`highWaterMark`][], but commonly, the default is the optimal 
+value for what system is running the application. In instances where you might 
+want to raise that value, go for it, but do so with caution!  
+
+## Build a WritableStream
+
+Let's extend the prototypical function of [`.write()`][]: 
```

Comment:
We are not extending `.write()` here, but extending `stream.Writable` with a custom `._write()` method, which is then _used_ internally by `.write()`.